lib/google/gax/errors.rb

```diff
@@ -27,20 +27,26 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+require 'English'
+
 module Google
   module Gax
     # Common base class for exceptions raised by GAX.
     class GaxError < StandardError
-      attr_reader :cause
-
       # @param msg [String] describes the error that occurred.
-      # @param cause [Error] the exception raised by a lower layer of
-      #   the RPC stack (for example, gRPC) that caused this
-      #   exception, or None if this exception originated in GAX.
-      def initialize(msg, cause: nil)
-        msg = "GaxError #{msg}, caused by #{cause}" if cause
+      def initialize(msg)
+        msg = "GaxError #{msg}"
+        msg += ", caused by #{$ERROR_INFO}" if $ERROR_INFO
         super(msg)
-        @cause = cause
+        @cause = $ERROR_INFO
+      end
+
+      # cause is a new method introduced in 2.1.0, bring this
+      # method if it does not exist.
+      unless respond_to?(:cause)
```

Comment:
The oldest supported Ruby version is 2.0.0 -- because gcloud-ruby has this (https://github.com/GoogleCloudPlatform/gcloud-ruby/blob/master/gcloud.gemspec#L19).

Modified .travis.yml to test with those old Rubies.